### PR TITLE
Implemented list deleted root blob with versions

### PR DESCRIPTION
**Pull request by gapra-msft**

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/models/ContainersGetPropertiesHeaders.java [hunk 1]
```diff
@@ -32,12 +32,6 @@
     @JsonProperty(value = "x-ms-version")
     private String xMsVersion;
 
-    /*
-     * The x-ms-immutable-storage-with-versioning-enabled property.
-     */
-    @JsonProperty(value = "x-ms-immutable-storage-with-versioning-enabled")
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/models/ContainersGetPropertiesHeaders.java:38]: I think this is due to swagger being not fully merged yet

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/models/ContainersGetPropertiesHeaders.java [hunk 2]
```diff
@@ -373,6 +350,26 @@ public ContainersGetPropertiesHeaders setXMsHasImmutabilityPolicy(Boolean xMsHas
         return this;
     }
 
+    /**
+     * Get the xMsVersionLevelWormEnabled property: The x-ms-version-level-worm-enabled property.
+     *
+     * @return the xMsVersionLevelWormEnabled value.
+     */
+    public Boolean isXMsVersionLevelWormEnabled() {
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/models/ContainersGetPropertiesHeaders.java:358]: This is due to swagger being out of date

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobContainerItemProperties.java [hunk 3]
```diff
@@ -348,25 +348,22 @@ public BlobContainerItemProperties setRemainingRetentionDays(Integer remainingRe
...
-     * Get the isImmutableStorageWithVersioningEnabled property: Indicates if version level worm is enabled on this
-     * container.
+     * Get the isVersionLevelWormEnabled property: Indicates if version level worm is enabled on this container.
      *
-     * @return the isImmutableStorageWithVersioningEnabled value.
+     * @return the isVersionLevelWormEnabled value.
      */
-    public Boolean isImmutableStorageWithVersioningEnabled() {
-        return this.isImmutableStorageWithVersioningEnabled;
+    public Boolean isVersionLevelWormEnabled() {
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobContainerItemProperties.java:355]: Swagger is out of date

File: sdk/storage/azure-storage-blob/swagger/README.md [hunk 4]
```diff
@@ -16,7 +16,7 @@ autorest --java --use:@autorest/java@4.0.x
 
 ### Code generation settings
 ``` yaml
-input-file: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/storage-dataplane-preview/specification/storage/data-plane/Microsoft.BlobStorage/preview/2020-10-02/blob.json
+input-file: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/a16ce6ff5dd8e57c4300b81dfa06021e13b942f5/specification/storage/data-plane/Microsoft.BlobStorage/preview/2020-10-02/blob.json
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-blob/swagger/README.md:19]: Undo when swagger merged

**Review by jaschrep-msft — APPROVED** [review]
Is there a way to get copy source auth out of here so we don't hit that merge conflict down the line?

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/ModelHelper.java [hunk 5]
```diff
@@ -265,6 +265,8 @@ public static BlobItem populateBlobItem(BlobItemInternal blobItemInternal) {
         blobItem.setObjectReplicationSourcePolicies(
             transformObjectReplicationMetadata(blobItemInternal.getObjectReplicationMetadata()));
 
+        blobItem.setHasVersionsOnly(blobItemInternal.isHasVersionsOnly());
```
Comment (kasobol-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/ModelHelper.java:268]: `isHasVersionsOnly` :scream: . The generator's code is very idiomatic.

**Review by kasobol-msft — APPROVED** [review]

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/ModelHelper.java (hunk 5 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/ModelHelper.java:268]: Yeah - it's kinda bad about booleans

**Pull request merged**